### PR TITLE
[Test] Regression Test for update_all and apply_edges

### PR DESCRIPTION
**Pull request by VoVAllen**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: benchmarks/benchmarks/api/bench_builtin_apply_edges.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import time
+import dgl
+import torch
+import numpy as np
+import dgl.function as fn
+
+from .. import utils
+
+
+@utils.benchmark('time', timeout=7200)
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_builtin_apply_edges.py:10]: This is very large. We should have a reasonable time estimation. Does feat_size=512 on livejournal takes too long?

File: benchmarks/benchmarks/api/bench_builtin_apply_edges.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    graph = utils.get_graph(graph_name, format)
+    # Remove format strict
+    graph = graph.formats(['coo', 'csr', 'csc'])
+    graph = graph.to(device)
+    graph.ndata['h'] = torch.randn(
+        (graph.num_nodes(), feat_size), device=device)
+
+    reduce_builtin_dict = {
+        'u->e': fn.copy_u('h', 'x'),
+        'v->e': fn.copy_u('h', 'x'),
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_builtin_apply_edges.py:26]: no need to test this

File: benchmarks/benchmarks/api/bench_builtin_apply_edges.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        'u+v': fn.u_add_v('h', 'h', 'x'),
+        'u*v': fn.u_mul_v('h', 'h', 'x'),
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_builtin_apply_edges.py:28]: only test u+v is fine

File: benchmarks/benchmarks/api/bench_builtin_update_all.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    graph = graph.formats(['coo', 'csr', 'csc'])
+    graph = graph.to(device)
+    graph.ndata['h'] = torch.randn(
+        (graph.num_nodes(), feat_size), device=device)
+    graph.edata['e'] = torch.randn(
+        (graph.num_edges(), feat_size), device=device)
+    
+    msg_builtin_dict = {
+        'copy_u': fn.copy_u('h', 'x'),
+        'copy_v': fn.copy_u('h', 'x'),
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_builtin_update_all.py:29]: no need to test this

**Comment by jermainewang** [comment]
Some of the tests are redundant. Consider removing them. Also, I'd like to add a graph that is similar to the block graph from neighbor sampling (degrees are similar).

**Comment by VoVAllen** [comment]
@jermainewang I addressed the comments and updated the `multi_update_all` tests. I found it will stuck when using asv run, so I added the `thread_wrapped_func`

File: benchmarks/benchmarks/api/bench_builtin_apply_edges.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+@utils.benchmark('time', timeout=600)
+@utils.parametrize('graph_name', ['cora', 'livejournal'])
+@utils.parametrize('format', ['coo', 'csr'])
+@utils.parametrize('feat_size', [8, 32, 128, 512])
+@utils.parametrize('reduce_type', ['u->e', 'u+v'])
+def track_time(graph_name, format, feat_size, reduce_type):
+    device = utils.get_bench_device()
+    graph = utils.get_graph(graph_name, format)
+    # Remove format strict
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_builtin_apply_edges.py:18]: remove format restriction

File: benchmarks/benchmarks/api/bench_builtin_apply_edges.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    graph = graph.formats(['coo', 'csr', 'csc'])
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_builtin_apply_edges.py:19]: you should extend `utils.get_graph` to allow multiple formats

File: benchmarks/benchmarks/api/bench_builtin_multi_update_all.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    device = utils.get_bench_device()
+    dd = {}
+    candidate_edges = [dgl.data.CoraGraphDataset(verbose=False)[0].edges(), dgl.data.PubmedGraphDataset(verbose=False)[
+        0].edges(), dgl.data.CiteseerGraphDataset(verbose=False)[0].edges()]
+    for i in range(num_relations):
+        dd[('n1', 'e_{}'.format(i), 'n2')] = candidate_edges[i %
+                                                             len(candidate_edges)]
+    graph = dgl.heterograph(dd)
+
+    # Remove format strict
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_builtin_multi_update_all.py:24]: the comment doesn't make sense

File: benchmarks/benchmarks/api/bench_builtin_multi_update_all.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        (graph.num_nodes('n2'), feat_size), device=device)
+
+    # dry run
+    update_dict = {}
+    for i in range(num_relations):
+        update_dict['e_{}'.format(i)] = (
+            fn.copy_src('h', 'm'), fn.sum('m', 'h'))
+    graph.multi_update_all(
+        update_dict,
+        "sum")
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_builtin_multi_update_all.py:38]: let's test one more case, "stack" cross-reducer.

File: benchmarks/benchmarks/api/bench_udf_multi_update_all.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+import numpy as np
+import dgl.function as fn
+
+
+from .. import utils
+
+
+@utils.benchmark('time', timeout=600)
+@utils.parametrize('feat_size', [32, 128, 512])
+@utils.parametrize('num_relations', [3, 6, 12])
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_udf_multi_update_all.py:14]: these configs are quite similar. use `[5, 50, 500]`

**Pull request merged**